src/gui/sdlmain.cpp

```diff
@@ -1692,6 +1775,44 @@ Bitu GFX_SetSize(int width,
 		if (rinfo.flags & SDL_RENDERER_ACCELERATED)
 			retFlags |= GFX_HARDWARE;
 
+		// Copied from the OpenGL path below; used to center texturepp output
+		int window_width  = 0;
+		int window_height = 0;
+		SDL_GetWindowSize(sdl.window, &window_width, &window_height);
+
+		const auto &desired_w = sdl.desktop.window.width;
+		const auto &desired_h = sdl.desktop.window.height;
+		const bool window_doesnt_match_desired = (desired_w != window_width ||
+		                                          desired_h != window_height);
+		const bool desired_size_is_valid = (desired_w > 0 && desired_h > 0);
+
+		// Adjust the window size if needed and permitted
+		if (sdl.scaling_mode != SCALING_MODE::PERFECT &&
+		    window_doesnt_match_desired && desired_size_is_valid &&
+		    !sdl.desktop.window.adjusted_initial_size) {
+			sdl.desktop.window.adjusted_initial_size = true;
+			safe_set_window_size(desired_w, desired_h);
+			SDL_GetWindowSize(sdl.window, &window_width, &window_height);
+		}
+
+		int canvas_width  = 0;
+		int canvas_height = 0;
```

Comment:
For consistency, suggest: `SDL_Rect canvas = ();`

Reply:
Then the introduction of "canvas" into the code will use the same type throughout and will read the same where ever its used.

(We're optimizing on consistency and readability, despite `SDL_Rect` not being a perfect skin-tight fit,  :sweat_smile: , but that can come later)

Reply:
Done.